pkg/integrations/prometheus/promql/metricsSearchHandler.go

```diff
@@ -562,11 +562,11 @@ func ProcessUiMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
 	metricQueriesList := make([]*structs.MetricsQuery, 0)
 	var timeRange *dtu.MetricsTimeRange
 	hashList := make([]uint64, 0)
-	for _, metricQuery := range metricQueryRequest {
-		hashList = append(hashList, metricQuery.MetricsQuery.HashedMName)
-		metricQueriesList = append(metricQueriesList, &metricQuery.MetricsQuery)
-		segment.LogMetricsQuery("PromQL metrics query parser", &metricQuery, qid)
-		timeRange = &metricQuery.TimeRange
+	for i := range metricQueryRequest {
+		hashList = append(hashList, metricQueryRequest[i].MetricsQuery.HashedMName)
+		metricQueriesList = append(metricQueriesList, &metricQueryRequest[i].MetricsQuery)
+		segment.LogMetricsQuery("PromQL metrics query parser", &metricQueryRequest[i], qid)
+		timeRange = &metricQueryRequest[i].TimeRange
```

Comment:
What's the purpose of this change?

Same comment for lines 738-742

Reply:
I think there is a bug in here. `var metricQueriesList []*structs.MetricsQuery`. 

In the past, the code is like:
```
for _, mQ := range metricQueryRequest {
......
metricQueriesList = append(metricQueriesList, &metricQuery.MetricsQuery)
}
```

If there are 2 metricQueries `a` and `b` in the metricQueryRequest. After first iteration, `metricQueriesList = {mQ.MetricsQuery}`. The references of `a` and `mQ` point to the same address. However, when it comes to the second iteration, `mQ` will point to b. Which will lead to previous `mQ.MetricsQuery` point to `b` as well. Please correct me if i am wrong.

Simple test:
```
	for _, metricQuery := range metricQueryRequest {
		metricQueriesList = append(metricQueriesList, &metricQuery.MetricsQuery)
		log.Error("fjl test1111 mQuery:", metricQuery.MetricsQuery.MetricName)
	}

	for _, mQuery := range metricQueriesList {
		log.Error("fjl test22222 mQuery:", mQuery.MetricName)
	}
```
![image](https://github.com/siglens/siglens/assets/67371917/29a06b13-9542-4516-9d6c-4d2a3a60957e)

Reply:
I'm not understanding that. Maybe you can show me sometime Tuesday